docusaurus/docs/tutorials/ui-extensions/create-an-extension-point.md

```diff
@@ -0,0 +1,218 @@
+    ```
+
+### Creating an extension point for links
+
+```tsx
+import { usePluginLinks } from '@grafana/runtime';
+
+export const InstanceToolbar = () => {
+  // The `extensionPointId` must be prefixed.
+  // - Core Grafana -> prefix with "grafana/"
+  // - Plugin       -> prefix with "{your-plugin-id}/"
+  //
+  // This is also what plugins use when they call `.addLink()`
+  const extensionPointId = 'myorg-foo-app/toolbar';
+  const { links, isLoading } = usePluginLinks({ extensionPointId });
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div>
+      {/* Loop through the links added by plugins */}
+      {links.map(({ id, title, path, onClick }) => (
+        <a href={path} title={title} key={id} onClick={onClick}>
+          {title}
+        </a>
+      ))}
+    </div>
+  );
+};
+```
+
+### Passing data to links
+
+```tsx
+import { usePluginLinks } from '@grafana/runtime';
+
+export const InstanceToolbar = ({ instanceId }) => {
+  const extensionPointId = 'myorg-foo-app/toolbar';
+  // Heads up! Always use `useMemo()` in case the `context` object has any "dynamic" properties
+  // to prevent unnecessary re-renders (Otherwise a new object would be created on every render, that could
+  // result in a new links{} object, that could trigger a new re-render, and so on.)
+  const context = useMemo(() => ({ instanceId }), [instanceId]);
+  const { links, isLoading } = usePluginLinks({ extensionPointId, context });
+
+  // ...
+};
+```
+
+### Limit the number of extensions by plugins
+
+You might have limited space on the UI and you would like the limit the number of extensions plugins can register to your extension point. **(By default there is no limit.)**
+
+```tsx
+import { usePluginLinks } from '@grafana/runtime';
+
+export const InstanceToolbar = () => {
+  // Only one link per plugin is allowed.
+  // (If a plugin registers more than one links, then the rest will be ignored
+  // and won't be returned by the hook.)
+  const { links, isLoading } = usePluginLinks({ extensionPointId, limitPerPlugin: 1 });
+
+  // ...
+};
+```
+
+### Limit which plugins can register links
+
+```tsx
+import { usePluginLinks } from '@grafana/runtime';
+
+export const InstanceToolbar = () => {
+  const { links, isLoading } = usePluginLinks({ extensionPointId, limitPerPlugin: 1 });
+
+  // You can rely on the `link.pluginId` prop to filter based on the plugin
+  // that has registered the extension.
+  const allowedLinks = useMemo(() => {
+    const allowedPluginIds = ['myorg-a-app', 'myorg-b-app'];
+    return links.filter(({ pluginId }) => allowedPluginIds.includes(pluginId));
+  }, [links]);
+
+  // ...
+};
+```
+
+## Components
+
+### Best practices for rendering components
+
+- **Make sure your UI controls the behaviour** <br /> Component extensions can render different layouts and can respond to various kind of user interactions. Make sure that your UI defines clear boundaries for rendering components defined by other plugins.
```

Comment:
An example would help here

Reply:
I think these best practices are great! Not much to add from my side.

Not sure I can think of a good example here. @mckn WDYT?

Reply:
Yeah, I guess we can have an example where we limit the number of items being displayed in the UI etc.